app/noScript.js

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+'use strict'
+
+const urlParse = require('url').parse
+const Filtering = require('./filtering')
+const AppStore = require('../js/stores/appStore')
+const siteSettings = require('../js/state/siteSettings')
+const siteUtil = require('../js/state/siteUtil')
+const ipcMain = require('electron').ipcMain
+const messages = require('../js/constants/messages')
+
+module.exports.resourceName = 'noScript'
+
+// Resources that should be temporarily allowed. True = allow once,
+// false = allow until restart.
+let temporarilyAllowed = {}
+
+function startNoScript () {
+  Filtering.registerHeadersReceivedFilteringCB(onHeadersReceived)
+  ipcMain.on(messages.TEMPORARY_ALLOW_SCRIPTS, (e, origin, allowOnce) => {
+    temporarilyAllowed[origin] = allowOnce
+  })
+}
+
+function onHeadersReceived (details) {
+  let result = { resourceName: module.exports.resourceName }
+  if (!Filtering.isResourceEnabled(module.exports.resourceName)) {
+    return result
+  }
+  // Ignore whitelisted URL schemes and non-applicable resource types
+  let parsed = urlParse(details.firstPartyUrl)
+  if (['about:', 'chrome:', 'chrome-extension:'].includes(parsed.protocol) || ['stylesheet', 'script', 'image'].includes(details.resourceType) || parsed.hostname === 'localhost') {
+    return result
+  }
+
+  let origin = siteUtil.getOrigin(details.firstPartyUrl)
```

Comment:
I think there's a minor problem here from what I'd expect.  For example on my website (brianbondy.com) I have a video in the first blog post.  If I accept once the video doesn't load, I think because the iframe has a different first party url.   I'm ok with landing this as is though but just post an issue for that to handle it after in a different way (whitelisting the whole frame once we can reliably determine frame from filtering functions)